rs/index/src/posting_list/combined_file.rs

```diff
@@ -132,7 +132,7 @@ impl FixedIndexFile {
         Ok(transmute_u8_to_slice::<f32>(slice))
     }
 
-    pub fn get_posting_list(&self, index: usize) -> Result<&[u64]> {
+    pub fn get_posting_list(&self, index: usize) -> Result<impl Iterator<Item = u64> + '_> {
```

Comment:
in the `File` class, `get_posting_list` probably should return a byte buffer (taken from start & offset). Then use the Decoder class to decode them and get the iterator instead.